x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/support/SecurityMigrationExecutor.java

```diff
@@ -100,6 +109,21 @@ private void applyOutstandingMigrations(AllocatedPersistentTask task, int curren
         }
     }
 
+    /**
+     * Refresh security index to make sure that docs that were migrated are visible to the next migration and to prevent version conflicts
+     * or unexpected behaviour by APIs relying on migrated docs.
+     */
+    private void refreshSecurityIndex(ActionListener<Void> listener) {
```

Comment:
This change was suggested by Albert and I think it would be nice to get in now that we have a second migration. See [this](https://github.com/elastic/elasticsearch/issues/110562) issue. It could be a separate PR.